File: refactor_db.ts
import fs from 'fs';

let content = fs.readFileSync('src/App.tsx', 'utf-8');

// Add imports
content = content.replace(/import \{ auth, db \} from '\.\/firebase';/g, "import { auth, db, col, dbDoc } from './firebase';");

// 1. doc(collection(db, 'name')) -> dbDoc('name')
content = content.replace(/doc\(collection\(db,\s*'([^']+)'\)\)/g, "dbDoc('$1')");

// 2. collection(db, 'name') -> col('name')
content = content.replace(/collection\(db,\s*'([^']+)'\)/g, "col('$1')");

// 3. doc(db, 'name', id) -> dbDoc('name', $2)
content = content.replace(/doc\(db,\s*'([^']+)',\s*([^)]+)\)/g, "dbDoc('$1', $2)");

fs.writeFileSync('src/App.tsx', content, 'utf-8');
console.log("Done refactoring App.tsx!");

File: refactor_db.py
import re

with open('src/App.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

# Add imports
content = re.sub(r"import \{ auth, db \} from \'\.\/firebase\';", "import { auth, db, col, dbDoc } from './firebase';", content)

# 1. doc(collection(db, 'name')) -> dbDoc('name')
content = re.sub(r"doc\(collection\(db,\s*'([^']+)'\)\)", r"dbDoc('\1')", content)

# 2. collection(db, 'name') -> col('name')
content = re.sub(r"collection\(db,\s*'([^']+)'\)", r"col('\1')", content)

# 3. doc(db, 'name', id) -> dbDoc('name', id)
content = re.sub(r"doc\(db,\s*'([^']+)',\s*([^)]+)\)", r"dbDoc('\1', \2)", content)

with open('src/App.tsx', 'w', encoding='utf-8') as f:
    f.write(content)
print("Done!")
